pkg/releaseutil/kind_sorter_test.go

```diff
@@ -304,29 +313,68 @@ func TestKindSorterForHooks(t *testing.T) {
 		},
 	}
 
-	for _, test := range []struct {
-		description string
-		order       KindSortOrder
-		expected    string
-	}{
-		{"install", InstallOrder, "acij"},
-		{"uninstall", UninstallOrder, "jica"},
+	installCase := testCase{
+		description: "install_order_wihtout_weight",
+		order:       InstallOrder,
+		expected:    "acij",
+		hooks:       hooksInstallUninstall,
+	}
+	uninstallCase := testCase{
+		description: "uninstall_order_wihtout_weight",
+		order:       UninstallOrder,
+		expected:    "jica",
+		hooks:       hooksInstallUninstall,
+	}
+
+	// compositeOrderCase is a test case to check that order priorities is applied correctly:
+	// First by weight, then by kind, finally by name.
+	compositeOrderCase := testCase{
+		description: "install_order_composite",
+		order:       InstallOrder,
+		expected:    "00_pod_01_sa_02_pod_03_job_04_job_",
+		hooks: []*release.Hook{
+			{
+				// Pod goes third by Kind order.
+				Name: "02_pod_",
+				Kind: "Pod",
+			},
+			{
+				// ServiceAccount goes second, as per Kind InstallOrder.
+				Name: "01_sa_",
```

Comment:
I would change the names to sa_01_, pod_02, even if it's less readable, then you will be testing the case we want, because this would still be ordered correctly in the previous code, since alphanumerically 01 goes before 02 but sa goes after pod.

Reply:
Good point, let me fix that :)

Reply:
Done!